spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplateAuditingTests.java

```diff
@@ -82,6 +84,20 @@ public void testModifiedPrevProperties() {
 		this.datastoreTemplate.saveAll(Collections.singletonList(testEntity));
 	}
 
+	@Test
+	public void testInTransaction() {
+		when(datastore.runInTransaction(any()))
```

Comment:
I see mock setup all done in `datastoreTemplate()`. Would it make sense to move it there?

Reply:
Personally, I prefer keeping the specific setup where it's actually used, and Mockito could complain about unnecessary setup in the other tests, depending on its configuration. However, I've tested moving the code and it works, so Mockito is not in strict mode and the code can be arranged however you prefer.

Reply:
I don't feel strongly either way. I'm also not sure it's good that the verification is outside the test method, but that's a problem for another time. :)

Reply:
OK, I think I'll leave it that way then, not specifying every possible behavior for all the tests.

I agree the verification in an `Answer` is brittle, which is when I first ran the test, without the added behavior on the mock, it passed: the `Datastore.put()` method was never called! We could extract an assertion method which would use an `ArgumentCaptor` to verify the call and its parameter, instead. However it's not the only occurence in the project of assertions in `Answer`s, so just fixing this one instance probably doesn't really make sense.

Reply:
Yeah, no need to address that here.